### PR TITLE
Split mau-virtualization test, clean REGRESSION variable and fix virt-manager login screen detection

### PR DESCRIPTION
**Pull request by pdostal**
Hello,

1) This PR separates the QAM virtualization XEN / KVM scenarios into several parts:
 * `VIRT_PART=install`
* `START_DIRECTLY_AFTER_TEST=install`:
   * `VIRT_PART=save_and_restore`
   * `VIRT_PART=hotplugging`
   * `VIRT_PART=guest_management`
   * `VIRT_PART=virtmanager`
   * `VIRT_PART=dom_metrics`

2) Another change is that the `REGRESSION` variable has now just either `xen` or `kvm` value.

3) There is also fix for `virt-manager` trying to re-open the guest console while waiting for the guest login screen.

- Related ticket: [poo#57521](https://progress.opensuse.org/issues/57521)
- Needles: No new needles
- Verification run: [KVM](http://pdostal-server.suse.cz/tests/5871#dependencies) [XEN](http://pdostal-server.suse.cz/tests/5876#dependencies) [gi-guest_developing-on-host_developing-kvm](https://openqa.suse.de/tests/3681190)

### CODE REVIEW

**Comment by pdostal** [comment]
Hello @alice-suse, can you or someone from your team take a look? This is the `START_DIRECTLY_AFTER_TEST` feature. Thank you very much

File: lib/xen.pm [hunk 1]
```diff
@@ -30,7 +30,7 @@ use version_utils 'is_sle';
 #   * extra parameters for virsh create / xl create
 
 our %guests = ();
-if (check_var("REGRESSION", "xen-hypervisor") || check_var("REGRESSION", "xen-client")) {
+if (check_var("REGRESSION", "xen")) {
```
Comment (alice-suse) [inline lib/xen.pm]: Good rename of these variables! Actually the variables, like xen-hypervisor, qemu-hypervisor, are also used in feature tests like internal/external snapshot and virtual network management, for condition checking. Suggest to keep them unchanged after this time's adaption :)

File: tests/boot/boot_from_pxe.pm [hunk 2]
```diff
@@ -155,23 +155,27 @@ sub run {
...
+                wait_still_screen 5;
+                type_string $args;
+                send_key 'ret';
+                assert_screen 'orthos-grub-boot-initrd', $ssh_vnc_wait_time;
+                $args = "boot";
+                type_string $args;
+                send_key "ret";
+                assert_screen $ssh_vnc_tag, $ssh_vnc_wait_time;
+            }
         }
```
Comment (alice-suse) [inline tests/boot/boot_from_pxe.pm]: What's the purpose for making the change here, line 159-176?

File: tests/boot/boot_from_pxe.pm [hunk 3]
```diff
@@ -155,23 +155,27 @@ sub run {
...
-            type_string $args;
-            send_key 'ret';
-            assert_screen 'orthos-grub-boot-initrd', $ssh_vnc_wait_time;
-            $args = "boot";
-            type_string $args;
-            send_key "ret";
-            assert_screen $ssh_vnc_tag, $ssh_vnc_wait_time;
+        # Proceed if the 'installation' console is ready
+        # otherwise the 'sol' console may be just freezed
+        if (check_screen(\@tags, $ssh_vnc_wait_time)) {
```
Comment (alice-suse) [inline tests/boot/boot_from_pxe.pm]: I am afraid , to let the original logic work, it should be assert_screen, or for ssh installation which needs to wait needle sshd-server-started show up , then select installation, will meet problem sometimes.

File: tests/virt_autotest/login_console.pm [hunk 4]
```diff
@@ -84,7 +84,15 @@ sub login_to_console {
...
-            send_key_until_needlematch("virttest-bootmenu-xen-kernel", 'down', 10, 5);
+
+            for (1 .. 20) {
+                if ($_ == 10) {
+                    reset_consoles;
+                    select_console 'sol', await_console => 0;
+                }
+                send_key 'down';
+                last if check_screen 'virttest-bootmenu-xen-kernel', 5;
+            }
```
Comment (alice-suse) [inline tests/virt_autotest/login_console.pm]: I guess this is to stablize login_console when sol console unstability happens. Cool! Would you mind adding some verification run to virtualization ipmi jobs , like https://openqa.nue.suse.com/tests/3661023?

**Comment by alice-suse** [comment]
> Hello @alice-suse, can you or someone from your team take a look? This is the `START_DIRECTLY_AFTER_TEST` feature. Thank you very much

Sure, thank you for involving me. Glad that you try this feature already. We have other higher priority things in hand, so we have to postpone this improvement until next year.

How do you see this feature? Is it helpful? Stable?

**Comment by pdostal** [comment]
> How do you see this feature? Is it helpful? Stable?

Yes, it requires more reboots but so far it works well for me.

File: lib/xen.pm (hunk 1 above)
Comment (pdostal) [reply lib/xen.pm]: We can support all variants. Is this solution OK?
```suggestion
if (get_var("REGRESSION") =~ "xen") {
```

File: tests/boot/boot_from_pxe.pm (hunk 2 above)
Comment (pdostal) [reply tests/boot/boot_from_pxe.pm]: This is another fix related to `sol` console instabilities. I only changed the line `158`, the rest is just indentation:
```diff
- assert_screen \@tags, $ssh_vnc_wait_time;
+ if (check_screen(\@tags, $ssh_vnc_wait_time)) { ... }
```

File: lib/xen.pm (hunk 1 above)
Comment (alice-suse) [reply lib/xen.pm]: I am fine with both. Thank you for being so considerate. You can decide which form to use :)

File: tests/boot/boot_from_pxe.pm (hunk 3 above)
Comment (alice-suse) [reply tests/boot/boot_from_pxe.pm]: Ok, get the point. So you want a try to reach installation console anyway to gain more success chance.

File: lib/xen.pm (hunk 1 above)
Comment (pdostal) [reply lib/xen.pm]: The new form should be more universal. I am already running verification run.

File: tests/virt_autotest/login_console.pm (hunk 4 above)
Comment (pdostal) [reply tests/virt_autotest/login_console.pm]: Hello @alice-suse, I was trying several times but I am not able to run that test in my environment. From what I can see it seems like our Prague PXE server does not have the required mounts so I would probably need to contact the infra team to fix it. Is it possible for you or someone from your team to just grab my branch and schedule a test run from your side?
obrazek

Comment (alice-suse) [reply tests/virt_autotest/login_console.pm]: @pdostal , you can use osd env to verify your PR to know whether it works on osd. 
Run "openqa-clone-custom-git-refspec https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9053 https://openqa.nue.suse.com/tests/3661023" on openqa.suse.de server is enough to trigger one verification job. For more, please refer to http://open.qa/docs/#_triggering_tests_based_on_an_any_remote_git_refspec_or_open_github_pull_request.

If you still have problem to do so, please let me know, I will help with it.

Comment (pdostal) [reply tests/virt_autotest/login_console.pm]: @alice-suse OK, I finally managed to do it. The [test run] seems to be OK.

```bash
$ openqa-clone-custom-git-refspec https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9053 https://openqa.suse.de/tests/3661023
Created job #3681190: sle-15-SP2-Online-x86_64-Build101.1-gi-guest_developing-on-host_developing-kvm@64bit-ipmi -> https://openqa.suse.de/t3681190
```

[test run]: https://openqa.suse.de/tests/3681190

Comment (alice-suse) [reply tests/virt_autotest/login_console.pm]: Cool! Thanks for it!

**Comment by alice-suse** [comment]
All LGTM. If no review from others today, I will merge it.

**Pull request merged**